packages/xrpl/src/models/transactions/setFee.ts

```diff
@@ -0,0 +1,57 @@
+import { BaseTransaction } from './common'
+
+export interface SetFeeBase extends BaseTransaction {
+  TransactionType: 'SetFee'
+}
+
+export interface SetFeePreAmendment extends SetFeeBase {
+  /**
+   * The charge, in drops of XRP, for the reference transaction, as hex. (This is the transaction cost before scaling for load.)
+   */
+  BaseFee: string
+  /**
+   * The cost, in fee units, of the reference transaction
+   */
+  ReferenceFeeUnits: number
+  /**
+   * The base reserve, in drops
+   */
+  ReserveBase: number
+  /**
+   * The incremental reserve, in drops
+   */
+  ReserveIncrement: number
+
+  BaseFeeDrops?: never
+  ReserveBaseDrops?: never
+  ReserveIncrementDrops?: never
```

Comment:
Can you add comments to these 3? - https://xrpl.org/feesettings.html (same below)

Reply:
Or a comment explaining why their type is set to `never`? - I thought these could be assigned values still?

Reply:
I do define them below on the type where they are set.  I can add a comment explaining that those fields will never be set before the amendment.

Reply:
I ended up not treating this like the same object and went with this style of documentation below.  The reason was that the previous way typedoc would interpret fields that are actual required in pre or post amendment as optional when in fact they are optional collectively.

![Monosnap SetFee | xrpl 2023-06-26 13-55-38](https://github.com/XRPLF/xrpl.js/assets/464895/262ae5af-7bd9-4b1e-a07a-43f2dc1a148e)